presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/TestAddIntermediateAggregations.java

```diff
@@ -250,15 +249,14 @@ public void testNoLocalParallel()
                 .on(p -> p.aggregation(af -> {
                     af.globalGrouping()
                             .step(AggregationNode.Step.FINAL)
-                            .addAggregation(p.variable("c"), expression("count(b)"), ImmutableList.of(BIGINT))
+                            .addAggregation(p.variable("c"), p.registerVariable(p.variable("b")).rowExpression("count(b)"))
                             .source(
                                     p.gatheringExchange(
                                             ExchangeNode.Scope.REMOTE_STREAMING,
                                             p.aggregation(ap -> ap.globalGrouping()
                                                     .step(AggregationNode.Step.PARTIAL)
-                                                    .addAggregation(p.variable("b"), expression("count(a)"), ImmutableList.of(BIGINT))
-                                                    .source(
-                                                            p.values(p.variable("a"))))));
+                                                    .source(p.values(p.variable("a")))
+                                                    .addAggregation(p.variable("b"), p.rowExpression("count(a)")))));
```

Comment:
Is there any reason for the source and AddAggregation reorder?

Reply:
Yeah, in some cases we want .variable() call to be registered before. This is needed to parse expressions containing variables.